### PR TITLE
Add `mean` and `variance` that take a Tensor<Int32> for `axes`.

### PR DESCRIPTION
**Pull request by bgogul**
This is needed to get tracing to work with batch norm.  See https://bugs.swift.org/browse/TF-406.

### CODE REVIEW

File: stdlib/public/TensorFlow/Ops.swift [hunk 1]
```diff
@@ -1292,6 +1292,19 @@ public extension Tensor where Scalar : Numeric {
...
+  /// Returns the arithmetic mean along the specified axes. The reduced
+  /// dimensions are retained with value 1.
+  /// - Parameter axes: The dimensions to reduce.
+  /// - Precondition: Each value in `axes` must be in the range `-rank..<rank`.
+  @inlinable @inline(__always)
+  @differentiable(
+    wrt: self, vjp: _vjpMean(alongAxes:)
+    where Scalar : TensorFlowFloatingPoint
+  )
+  func mean(alongAxes axes: Tensor<Int32>) -> Tensor {
```
Comment (bgogul) [inline stdlib/public/TensorFlow/Ops.swift]: @rxwei, would it make sense to change mean(alongAxes axes: [Int32]) to call this directly?

**Comment by bgogul** [comment]
@swift-ci please test tensorflow

File: stdlib/public/TensorFlow/Ops.swift (hunk 1 above)
Comment (dan-zheng) [reply stdlib/public/TensorFlow/Ops.swift]: Weighing in: yes, I think that would make sense.

File: stdlib/public/TensorFlow/Gradients.swift [hunk 2]
```diff
@@ -589,6 +589,15 @@ extension Tensor where Scalar : TensorFlowFloatingPoint {
     })
   }
 
+  @inlinable
+  func _vjpMean(alongAxes axes: Tensor<Int32>) -> (Tensor, (Tensor) -> Tensor) {
+    let value = mean(alongAxes: axes)
+    return (value, { [shape = shapeTensor,
+            count = Raw.gather(params: shapeTensor, indices: axes).product()] in
```
Comment (dan-zheng) [inline stdlib/public/TensorFlow/Gradients.swift]: Could you please declare `count` as a local variable under `let value`?

**Review by dan-zheng — APPROVED** [review]

File: stdlib/public/TensorFlow/Ops.swift (hunk 1 above)
Comment (bgogul) [reply stdlib/public/TensorFlow/Ops.swift]: Done.

File: stdlib/public/TensorFlow/Gradients.swift (hunk 2 above)
Comment (bgogul) [reply stdlib/public/TensorFlow/Gradients.swift]: Done.

**Comment by bgogul** [comment]
@swift-ci please test tensorflow

File: stdlib/public/TensorFlow/Ops.swift (hunk 1 above)
Comment (rxwei) [reply stdlib/public/TensorFlow/Ops.swift]: Yep, Dan's right.

**Review by rxwei — APPROVED** [review]

**Comment by rxwei** [comment]
Have you run tests?

**Comment by bgogul** [comment]
> Have you run tests?

yes. 

https://ci-external.swift.org/view/Pull%20Request/job/swift-PR-TensorFlow-Linux/2006/
https://ci-external.swift.org/view/Pull%20Request/job/swift-PR-TensorFlow-Linux-gpu/1222/
https://ci-external.swift.org/view/Pull%20Request/job/swift-PR-TensorFlow-macOS/1301/
